### DISCUSSION TITLE
Load a custom robot given the .3dm file

### DISCUSSION THREAD

**Discussion by Jishnu8** [visose/Robots#221] [Q&A] [answered]
Hi guys,
I would like to load a custom robot using the robots plugin. I have the .3dm file of the robot I want to load. Is this possible using the robots plugin?

**Comment by visose** [comment]
If it's one from one of the supported manufacturers, and you have created a .3dm/.xml files formatted correctly for the plugin, then sure.

**Comment by Jishnu8** [comment]
Hi @visose ,
PFA the link to the .3dm file of the robot I would like to load using the plugin. How do I go about creating the .xml file to load the robot?

["KR300 R2500"](https://drive.google.com/drive/folders/1el24e9btrRd3ZTTbHvONVNEVzwyMn1eE)

**Reply by visose** [reply]
Please check: https://github.com/visose/Robots/wiki/Robot-libraries
Most people are able to create an .xml by downloading a library that contains a similar robot and adjusting the necessary values. You can search this the forum if you get stuck, it's likely the question will have already been asked. If it's an original question feel free to ask it then.

**Comment by visose** [comment, marked as answer]
A `.3dm` file by itself is not enough to load a custom robot. The robot must be from a supported manufacturer/kinematic family and needs a matching `.xml` describing the robot cell, joints, IO, tools, etc. The usual workflow is to copy a similar library from the `libraries` branch, edit the XML values and top-level layer names, and keep the `.xml`/`.3dm` pair together in the Robots library folder.
